### PR TITLE
e2e tests

### PR DESCRIPTION
**Pull request by jakemcdermott**
## AWX E2E
```shell
# setup
docker exec -i tools_awx_1 sh <<-EOSH
  awx-manage createsuperuser --noinput --username=awx-e2e --email=a@wx.com
  awx-manage update_password --username=awx-e2e --password=password
  make --directory=/awx_devel DATA_GEN_PRESET=e2e bulk_data
EOSH

# run with with a live browser
npm --prefix awx/ui run e2e -- --env=debug
```

### CODE REVIEW

File: awx/ui/client/test/e2e/commands/login.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    this.api.page.login()
+        .navigate()
+        .waitForElementVisible('@submit', this.api.globals.longWait)
+        .waitForElementNotVisible('div.spinny')
+        .setValue('@username', username)
+        .setValue('@password', password)
+        .click('@submit')
+        .waitForElementVisible('div.spinny')
+        // temporary hack while layout issue is being resolved
```
Comment (jakemcdermott) [inline awx/ui/client/test/e2e/commands/login.js]: https://github.com/ansible/awx/issues/52

File: awx/ui/client/test/e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## AWX E2E
```
Comment (gconsidine) [inline awx/ui/client/test/e2e/README.md:1]: It's implied with the `docker exec` command below, but might be worth explicitly mentioning the API server must be up and running on `https://localhost:8043` (if not overridden) and what can be overridden with the environment variables referenced in the settings file.

File: awx/ui/client/test/e2e/commands/inject.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+exports.command = function(deps, script, callback) {
+    let self = this;
```
Comment (gconsidine) [inline awx/ui/client/test/e2e/commands/inject.js]: There's a mix of `self` and `this` -- are there different object contexts at play here?

File: awx/ui/client/test/e2e/settings.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    retryAssertionTimeout: AWX_E2E_TIMEOUT_MEDIUM,
+    selenium_host: AWX_E2E_SELENIUM_HOST,
+    selenium_port: AWX_E2E_SELENIUM_PORT,
+    shortTimeout: AWX_E2E_TIMEOUT_SHORT,
+    waitForConditionTimeout: AWX_E2E_TIMEOUT_MEDIUM,
+    test_workers: {
+        enabled: (AWX_E2E_WORKERS > 0),
+        workers: AWX_E2E_WORKERS
+    },
+    before(done) {
```
Comment (gconsidine) [inline awx/ui/client/test/e2e/settings.js]: Could these before/after calls be removed?

File: awx/ui/client/test/e2e/settings.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const AWX_E2E_URL            = process.env.AWX_E2E_URL            || 'https://localhost:8043';
```
Comment (gconsidine) [inline awx/ui/client/test/e2e/settings.js]: I'd vote for not vertically aligning anything, but that's just a personal preference.

File: awx/ui/client/test/e2e/tests/test-credentials-read-only.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+            let { adminCredential, auditor } = store;
+
+            return $http.get('/api/v2/me')
+                .then(({ data }) => {
+                    let resource = 'Amazon%20Web%20Services+cloud';
+                    adminCredential.user = data.results[0].id;
+
+                    return $http.get(`/api/v2/credential_types/${resource}`);
+                }).then(({ data }) => {
```
Comment (gconsidine) [inline awx/ui/client/test/e2e/tests/test-credentials-read-only.js]: I prefer `.then` on a new line like you have on line 42. It's not a strong preference, but it's what is already in place in the refactored `lib/` and `features/` code.

**Review by gconsidine — APPROVED** [review]
This looks great! I'm excited about having a solid e2e test foundation in place.

There's the one race-type condition on the credential-type test that we discovered together. Aside from that, I'd say bring in the latest on `devel` ahead of merging and run the tests one more time.

If they pass and the other checks pass, fire away!

Nice work.

File: awx/ui/client/test/e2e/settings.js (hunk 5 above)
Comment (jakemcdermott) [reply awx/ui/client/test/e2e/settings.js]: sounds good to me - done

File: awx/ui/client/test/e2e/settings.js (hunk 4 above)
Comment (jakemcdermott) [reply awx/ui/client/test/e2e/settings.js]: yep - removed

File: awx/ui/client/test/e2e/commands/inject.js (hunk 3 above)
Comment (jakemcdermott) [reply awx/ui/client/test/e2e/commands/inject.js]: This used to be the case but no longer. Removed.

File: awx/ui/client/test/e2e/tests/test-credentials-read-only.js (hunk 6 above)
Comment (jakemcdermott) [reply awx/ui/client/test/e2e/tests/test-credentials-read-only.js]: I have no preference here and so will default to your preference. Done :)

File: tools/data_generators/rbac_dummy_data_generator.py [hunk 7]
```diff
@@ -107,6 +107,9 @@
 
     options.update({cols[0]: cols[col + 1] for cols in split_lines})
 
+    if not options['prefix']:
```
Comment (jakemcdermott) [inline tools/data_generators/rbac_dummy_data_generator.py:110]: confirmed with @AlanCoding yesterday that these changes make sense

File: awx/ui/client/test/e2e/commands/login.js (hunk 1 above)
Comment (jakemcdermott) [reply awx/ui/client/test/e2e/commands/login.js]: removed this PR now that a fix for #52 has landed.

**Comment by mabashian** [comment]
I haven't looked at the code yet but I did pull this down and try it out successfully.  Great job!  Looking forward to building this out in the future.

**Pull request merged**